includes/wp-cli.php

```diff
@@ -163,15 +165,252 @@ function import_from_csv( $args, $assoc_args ) {
+					$redirect['to']['formatted'] = $redirect_to_validate->post_excerpt;
+				} else {
+					$redirect['to']['raw'] = $redirect_to_validate->post_parent;
+					$redirect['destination_type'] = 'post';
+				}
+
+				// Validate the redirect.
+				$validate = WPCOM_Legacy_Redirector::validate_redirect( $redirect, $post_types );
+
+				if ( true !== $validate ) {
+					$notices[] = $validate;
+					if ( 'draft' !== $redirect['post_status'] ) {
+						$update_redirect_status['draft'][] = $redirect['id'];
+					}
+					continue;
+				}
+
+				if ( 'post' === $redirect['destination_type'] ) {
+
+					// Don't verify urls to validated post ids unless the [--strict] flag is explicitly set.
+					if ( ! $assoc_args['strict'] ) {
+						if ( 'publish' !== $redirect['post_status'] ) {
+							$update_redirect_status['publish'][] = $redirect->ID;
+						}
+						continue;
+					}
+
+					// Reuse parent post object in case of multiple redirects to same post ID.
+					if ( ! isset( $parent->ID ) || intval( $redirect['parent']['id'] ) !== $parent->ID ) {
+						$parent = get_post( $redirect['parent']['id'] );
+					}
+
+					$redirect['to']['formatted'] = get_permalink( $parent );
+				}
+
+				// Ping the URL and get the headers
+				$redirect_head = wp_remote_head( $redirect['from']['url'] );
+				if ( is_wp_error( $redirect_head ) ) {
+					$notices[] = array(
+						'id'        => $redirect['id'],
+						'from_url'  => $redirect['from']['path'],
+						'to_url'    => $redirect['to']['raw'],
+						'message'   => implode( "\n", $redirect_head->get_error_messages() ),
+					);
+					continue;
+				}
+				$redirect['redirect']['resulting_url'] = wp_remote_retrieve_header( $redirect_head, 'location' );
+				$redirect['redirect']['status'] = wp_remote_retrieve_response_code( $redirect_head );
+				$redirect['redirect']['status_msg'] = wp_remote_retrieve_response_message( $redirect_head );
+
+				$verify = WPCOM_Legacy_Redirector::verify_redirect( $redirect );
+
+				if ( true === $verify ) {
+					if ( $assoc_args['verbose'] ) {
+						$notices[] = array(
+							'id'        => $redirect['id'],
+							'from_url'  => $redirect['from']['path'],
+							'to_url'    => $redirect['to']['raw'],
+							'message'   => 'Verified',
+						);
+					}
+
+					if ( 'publish' !== $redirect['post_status'] ) {
+						$update_redirect_status['publish'][] = $redirect['id'];
+					}
+				} else {
+					$notices[] = $verify;
+					if ( 'draft' !== $redirect['post_status'] ) {
+						$update_redirect_status['draft'][] = $redirect['id'];
+					}
+					continue;
+				}
+			} // End foreach().
+
+			// Update redirect status
+			if ( count( $update_redirect_status ) > 0 ) {
+				foreach ( $update_redirect_status as $redirect_status => $redirects_to_update ) {
+					foreach ( $redirects_to_update as $redirect_to_update ) {
+						$updated_rows = $wpdb->update( $wpdb->posts, array( 'post_status' => $redirect_status ), array( 'ID' => $redirect_to_update ) );
```

Comment:
We'll need to trigger a `clean_post_cache` for these posts otherwise we'll end up have stale values that are mismatched.